src/components/Documents/DocumentCard.jsx

```diff
@@ -0,0 +1,233 @@
+ */
+
+/**
+ * DocumentCard - A component that displays a single document's information
+ * and provides actions (preview, share, delete) through a context menu.
+ *
+ * @memberof Documents
+ * @name DocumentCard
+ * @param {object} props - Component props
+ * @param {Document} props.document - The document object to be displayed
+ * @param {Function} props.onPreview - Function to handle the "Preview" action
+ * @param {Function} props.onShare - Function to handle the "Share" action
+ * @param {Function} props.onDelete - Function to handle the "Delete" action
+ * @returns {React.JSX.Element} The DocumentCard component
+ */
+const DocumentCard = ({ document, onShare, onDelete, onPreview }) => {
+  const theme = useTheme();
+
+  const [anchorEl, setAnchorEl] = useState(null);
+  const [openMenu, setOpenMenu] = useState(null);
+
+  // Opens the context menu
+  const handleClick = (event, clickedDocument) => {
+    setAnchorEl(event.currentTarget);
+    setOpenMenu(clickedDocument.id);
+  };
+
+  // Closes the context menu
+  const handleClose = () => {
+    setAnchorEl(null);
+    setOpenMenu(null);
+  };
+
+  // Handles the click on a context menu item, executing the corresponding action
+  const handleMenuItemClick = (action, clickedDocument) => () => {
+    action(clickedDocument);
+    handleClose();
+  };
+
+  // ... (Styling definitions below)
+
+  const rowStyling = {
+    display: 'flex',
+    columnGap: '10px'
+  };
+
+  const columnStyling = {
+    display: 'flex',
+    flexDirection: 'column'
+  };
+
+  const dataStyling = {
+    ...rowStyling,
+    color: 'text.secondary',
+    fontSize: '.875rem',
+    justifyContent: 'flex-start'
+  };
+
+  const dataSectionStyling = {
+    ...columnStyling,
+    '& div': {
+      ...dataStyling
+    }
+  };
+
+  const dataColumnStyling = {
+    ...columnStyling,
+    flex: '1 1 90%',
+    rowGap: '5px',
+    '& div': {
+      ...dataSectionStyling
+    }
+  };
+
+  const actionsColumnStyling = {
+    ...columnStyling,
+    flex: '0 1 10%',
+    maxWidth: '54px',
+    justifyContent: 'center',
+    alignItems: 'flex-end',
+    marginRight: '-10px'
+  };
+
+  const cardStyling = {
+    marginBottom: '15px'
+  };
+
+  const cardContentStyling = {
+    ...rowStyling,
+    justifyContent: 'space-between'
+  };
+
+  // icon styles
+  const iconSize = {
+    height: '24px',
+    width: '24px'
+  };
+
+  const iconStyling = {
+    width: '100%'
+  };
+
+  return (
+    <Box>
+      <Card sx={cardStyling}>
+        <CardContent sx={cardContentStyling}>
+          <Box sx={dataColumnStyling}>
+            <Box>
+              <Typography
+                sx={{
+                  color: `${theme.palette.primary.text} !important`,
+                  fontSize: '1rem !important'
+                }}
+                variant="body1"
+                component="div"
+                noWrap
+              >
+                {truncateText(document.name) || '[No Name]'}
```

Comment:
I'm not sure truncateText is optimal here, certainly not with the default amount. It cuts off too quickly.

![2024-08-31](https://github.com/user-attachments/assets/5042ccf3-a457-427a-83b3-fc0c4d6aac47)

The Contacts table uses `maxWidth: '90%'` instead.

Reply:
Fair enough, I'll adjust that